main.py

```diff
@@ -0,0 +1,146 @@
+                find_and_label_all_revision_ids(gerrit, patchset)
+                upload_all_comments(gerrit, patchset)
+            except ValueError:
+                print(f'Failed to find {message.id} in index.')
+                pass
+            else:
+                print(f'Failed to find {message.id} in index.')
+                pass
+        remove_files(FILE_DIR)
+    else:
+        last_hash = message_dao.get_lash_hash()
+    
+    # Start the main server loop
+    while(True):
+        lash_hash = fill_message_directory(GIT_PATH, FILE_DIR, last_hash)
+        new_messages = archive_index.update(FILE_DIR)
+        
+        # Differentiate between messages to upload and comments
+        messages_to_upload : List[str] = []
+        messages_with_new_comments : Set[str] = set()
+        parent_patches : Set[str] = set()
+        parents, replies = split_parent_and_reply_messages(new_messages)
+        
+        # First separate between parents and replies. All parents of patchsets will be uploaded
+        for message in parents:
+            parent_patches.add(message.id)
+            messages_to_upload.append(message.id)
+        
+        # Determine which of the replies should be uploaded
+        for message in replies:
+            if message.in_reply_to in parent_patches:
+                continue
+            
+            # Reply is a patch to be uploaded (as the parent of patchset is not in new_messages)
+            if message.is_patch():
+                messages_to_upload.append(message.id)
+            # Reply is a comment that's parent is not in this batch of messages. It's parent's comments should be reuploaded
+            else:
+                messages_with_new_comments.add(message.in_reply_to)
+                
+        # Start uploading files
+        for message_id in messages_to_upload:
+            try:
+                email_thread = archive_index.find(message_id)
+                patchset = parse_comments(email_thread)
+                gerrit_git.apply_patchset_and_cleanup(patchset)
+                find_and_label_all_revision_ids(gerrit, patchset)
+                upload_all_comments(gerrit, patchset)
+            except ValueError:
+                print(f'Failed to find {message_id} in index.')
+                pass
+            else:
+                print(f'Failed to find {message_id} in index.')
+                pass
+            
+        for message_id in messages_with_new_comments:
+            try:
+                email_thread = archive_index.find(message_id)
+                patchset = parse_comments(email_thread)
+                upload_all_comments(gerrit, patchset)
+            except ValueError:
+                print(f'Failed to find {message_id} in index.')
+                pass
+            else:
+                print(f'Failed to find {message_id} in index.')
+                pass
+        
+        # Clean up directory for next loop
+        remove_files(FILE_DIR)
+        time.sleep(10)
```

Comment:
This method is  fairly huge. Can we maybe split it up into some logical helper functions?